### PR TITLE
[PT-141] - AddressBookTab

### PR DESCRIPTION
**Pull request by ItsFaizan**
# [PT-141](https://0xbase.atlassian.net/browse/PT-141?atlOrigin=eyJpIjoiM2EyZmVmNDJiZWQyNDcyNWE4MTUwYjg0ZTFmNDk5YjYiLCJwIjoiaiJ9) - AddressBookTab

### AddressBookTab created and wallet type identified plus bundles and addresses stored in local storage

[PT-141]: https://0xbase.atlassian.net/browse/PT-141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/containers/walletForm/WalletForm.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  const handleAddAddress = () => {
+    if (!newAddress) {
+      alert('Please enter an address.')
+      return
+    }
+
+    const walletType = identifyWalletType(newAddress)
+    if (!walletType) {
+      alert('Address not recognized as a valid wallet address.')
```
Comment (fuddin95) [inline src/containers/walletForm/WalletForm.tsx:46]: please remove `alert` anywhere in the code and we can either 
1. use modal or
2. turn the input field to red with incorrect wallet address type 
3. display a pop up

File: src/utils/identifyWalletType.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export function identifyWalletType(walletAddress: string): string | null {
```
Comment (fuddin95) [inline src/utils/identifyWalletType.ts:1]: Rename function to `getBlockchainType` as well as the file name

File: src/utils/identifyWalletType.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  const bitcoinRegex = /^(bc1|[13])[a-zA-HJ-NP-Z0-9]{25,39}$/
+  const ethereumRegex = /^0x[a-fA-F0-9]{40}$/
+  const solanaRegex = /^([1-9A-HJ-NP-Za-km-z]{44})$/
```
Comment (fuddin95) [inline src/utils/identifyWalletType.ts:4]: These should be moved out of the function maybe defined at the top of the file with the naming convention 
`bitcoinRegex` -> `BITCOIN_REGEX`

File: src/views/WalletDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+          variant="outlined"
+          value={newAddress}
+          onChange={(e) => setNewAddress(e.target.value)}
+          placeholder="Enter wallet address"
+        />
+        <Button
+          variant="contained"
+          sx={{
+            backgroundColor: '#1e3c31',
+            color: '#2cb84c',
```
Comment (fuddin95) [inline src/views/WalletDialog.tsx:79]: color needs to come from theme color

File: src/containers/walletForm/WalletForm.tsx (hunk 1 above)
Comment (ItsFaizan) [reply src/containers/walletForm/WalletForm.tsx:46]: Done input field turns red

File: src/views/WalletDialog.tsx (hunk 4 above)
Comment (ItsFaizan) [reply src/views/WalletDialog.tsx:79]: This is for button

File: src/utils/identifyWalletType.ts (hunk 2 above)
Comment (ItsFaizan) [reply src/utils/identifyWalletType.ts:1]: Done

**Review by fuddin95 — APPROVED** [review]

**Review by fuddin95 — APPROVED** [review]

**Pull request merged**